numpy/_array_api/_statistical_functions.py

```diff
@@ -0,0 +1,32 @@
+from __future__ import annotations
+
+from ._array_object import ndarray
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING:
+    from ._types import Optional, Tuple, Union, array
+
+import numpy as np
+
+def max(x: array, /, *, axis: Optional[Union[int, Tuple[int, ...]]] = None, keepdims: bool = False) -> array:
+    return ndarray._new(np.max(x._array, axis=axis, keepdims=keepdims))
+
+def mean(x: array, /, *, axis: Optional[Union[int, Tuple[int, ...]]] = None, keepdims: bool = False) -> array:
+    return ndarray._new(np.asarray(np.mean(x._array, axis=axis, keepdims=keepdims)))
+
+def min(x: array, /, *, axis: Optional[Union[int, Tuple[int, ...]]] = None, keepdims: bool = False) -> array:
+    return ndarray._new(np.min(x._array, axis=axis, keepdims=keepdims))
+
+def prod(x: array, /, *, axis: Optional[Union[int, Tuple[int, ...]]] = None, keepdims: bool = False) -> array:
+    return ndarray._new(np.asarray(np.prod(x._array, axis=axis, keepdims=keepdims)))
```

Comment:
My tests are still WIP and I haven't gotten to this file. Although this particular question is still being discussed a bit in the spec, see https://github.com/data-apis/array-api/pull/238.

Reply:
Fair, enough, I guess. Since your data-api's tests should notice the difference.  Note that the discussion there seems to forget that the default type is not the highest precision type.

Reply:
Also, it may be that it need fixing either way, since NumPy only adjusts bools and integers, not floats (although I am not sure what the argument for upcasting floats by default is, to be honest.)